sequencer/src/main/java/net/consensys/linea/sequencer/liveness/LivenessPlugin.java

```diff
@@ -0,0 +1,380 @@
+    }
+
+    log.info("{} stopped", PLUGIN_NAME);
+  }
+
+  @Override
+  public void onBlockAdded(AddedBlockContext addedBlockContext) {
+    if (!isPluginEnabled) return;
+
+    BlockHeader newBlock = addedBlockContext.getBlockHeader();
+    lastProcessedBlock.set(newBlock);
+
+    // Reset the last reported timestamp when a new block is added
+    lastReportedTimestamp.set(newBlock.getTimestamp());
+
+    log.debug(
+        "New block added: number={}, timestamp={}", newBlock.getNumber(), newBlock.getTimestamp());
+  }
+
+  private void checkBlockTimestampAndReport() {
+    if (!isPluginEnabled) return;
+
+    try {
+      BlockHeader lastBlock = lastProcessedBlock.get();
+      if (lastBlock == null) {
+        log.warn("No blocks available in the blockchain");
+        return;
+      }
+
+      long currentTimestamp = Instant.now().getEpochSecond();
+      long lastBlockTimestamp = lastBlock.getTimestamp();
+      long timeSinceLastBlock = currentTimestamp - lastBlockTimestamp;
+
+      log.debug(
+          "Checking block timestamp: lastBlockNumber={}, lastBlockTimestamp={}, currentTimestamp={}, timeSinceLastBlock={}s",
+          lastBlock.getNumber(),
+          lastBlockTimestamp,
+          currentTimestamp,
+          timeSinceLastBlock);
+
+      // Check if we need to report downtime
+      if (timeSinceLastBlock > maxBlockAgeSeconds) {
+        // Only report if we haven't reported recently or if significant time has passed
+        long timeSinceLastReport = currentTimestamp - lastReportedTimestamp.get();
+        if (timeSinceLastReport > maxBlockAgeSeconds) {
+          log.info(
+              "Sequencer appears to be down: lastBlockNumber={}, lastBlockTimestamp={}, timeSinceLastBlock={}s",
+              lastBlock.getNumber(),
+              lastBlockTimestamp,
+              timeSinceLastBlock);
+
+          // TODO: make sure these transaction are submitted first, in the first block created
+          sendSequencerUptimeTransaction(lastBlockTimestamp, currentTimestamp);
+          lastReportedTimestamp.set(currentTimestamp);
+        }
+      }
+    } catch (Exception e) {
+      log.error("Error in checkBlockTimestampAndReport", e);
+    }
+  }
+
+  private void sendSequencerUptimeTransaction(long lastBlockTimestamp, long currentTimestamp) {
+    try {
+      log.info(
+          "Sending sequencer uptime transaction: lastBlockTimestamp={}, currentTimestamp={}",
+          lastBlockTimestamp,
+          currentTimestamp);
+
+      // First call: updateStatus(false, lastBlockTimestamp)
+      Function functionDown =
+          new Function(
+              "updateStatus",
+              Arrays.asList(new Bool(false), new Uint256(lastBlockTimestamp)),
+              Collections.emptyList());
+
+      String encodedFunctionDown = FunctionEncoder.encode(functionDown);
+      byte[] callDataBytesDown = Numeric.hexStringToByteArray(encodedFunctionDown);
+      Bytes callDataDown = Bytes.wrap(callDataBytesDown);
```

Comment:
I don't know what the standard/dev guidelines are here, but some code pieces could be inlined - e.g. `functionDown` is only ever used in the `.encode`